### PR TITLE
Update travis to use both Ubuntu 20.04 and 18.04

### PR DESCRIPTION
**Pull request by Jellyfrog**
This gives us testing of both MySQL 5.7 and 8
Also bumps python to 3.6 and 3.8

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/functions.php [hunk 1]
```diff
@@ -2324,6 +2324,10 @@ function dump_db_schema($connection = null)
             $constraint_count = preg_match_all($constraint_regex, $create->{'Create Table'}, $constraints);
             for ($i = 0; $i < $constraint_count; $i++) {
                 $constraint_name = $constraints['name'][$i];
+
+                // MySQL 8 fix, remove RESTRICT from extra columns, it's the default.
+                $constraints['extra'][$i] = trim(str_replace(["ON UPDATE RESTRICT", "ON DELETE RESTRICT"], null, $constraints['extra'][$i]));
```
Comment (Jellyfrog) [inline includes/functions.php:2329]: not sure about this change yet.

Comment (murrant) [reply includes/functions.php:2329]: seams reasonable...

**Comment by murrant** [comment]
Looks like another net-snmp change in behavior.

**Comment by Jellyfrog** [comment]
This seems to be a bigger problem then just some changed behavior
```diff
OS ird: Discovered os data does not match that found in tests/data/ird.json
460Failed asserting that two arrays are equal.
461--- Expected
462+++ Actual
463@@ @@
464         0 => Array (
465             'sysName' => '<private>'
466             'sysObjectID' => '.1.3.6.1.4.1.2021.250.10'
467-            'sysDescr' => 'Linux DCH-4000P 2.4.20-uc0 #386 2013... 10... 24... ......... 14:08:33 CST armv3l'
468+            'sysDescr' => 'Linux DCH-4000P 2.4.20-uc0 #386 2013年 10月 24日 星期四 14:08:33 CST armv3l'
469             'sysContact' => null
470             'version' => null
471             'hardware' => null
```

**Comment by murrant** [comment]
Will need to be redone for actions

**Pull request closed without merging**